### PR TITLE
ARROW-3912: [Plasma][GLib] Add support for creating and referring objects

### PR DESCRIPTION
**Pull request by kou**

### CODE REVIEW

File: c_glib/plasma-glib/client.cpp [hunk 1]
```diff
@@ -21,47 +21,192 @@
...
+  auto priv = GPLASMA_CLIENT_CREATE_OPTIONS_GET_PRIVATE(options);
+  if (size) {
+    *size = priv->metadata_size;
+  }
+  return priv->metadata;
+}
+
 typedef struct GPlasmaClientPrivate_ {
-  std::shared_ptr<plasma::PlasmaClient> client;
+  plasma::PlasmaClient *client;
```
Comment (shiro615) [inline c_glib/plasma-glib/client.cpp]: Is that because this change prevent a memory leak?

File: c_glib/plasma-glib/client.cpp [hunk 2]
```diff
@@ -122,10 +267,160 @@ GPlasmaClient *
...
+  std::vector<plasma::ObjectBuffer> plasma_object_buffers;
+  auto status = plasma_client->Get(plasma_ids,
+                                   timeout_ms,
+                                   &plasma_object_buffers);
+  if (garrow_error_check(error, status, context)) {
+    auto plasma_object_buffer = plasma_object_buffers[0];
+    auto plasma_data = plasma_object_buffer.data;
+    auto plasma_metadata = plasma_object_buffer.metadata;
+    GArrowBuffer *data;
+    GArrowBuffer *metadata;
```
Comment (shiro615) [inline c_glib/plasma-glib/client.cpp]: Do we need to silence the warning?
https://travis-ci.org/apache/arrow/jobs/461631864#L3955

File: c_glib/plasma-glib/object.cpp [hunk 3]
```diff
@@ -0,0 +1,537 @@
...
+  if (!gplasma_object_check_not_released(priv, error, context)) {
+    return FALSE;
+  }
+
+  auto plasma_client = gplasma_client_get_raw(priv->client);
+  auto id_priv = GPLASMA_OBJECT_ID_GET_PRIVATE(priv->id);
+  auto status = plasma_client->Seal(id_priv->id);
+  auto success = garrow_error_check(error, status, context);
+  if (success) {
+    plasma_client->Release(id_priv->id);
```
Comment (shiro615) [inline c_glib/plasma-glib/object.cpp]: Do we need to receive `status` to silence the warning?
https://travis-ci.org/apache/arrow/jobs/461631864#L3940

File: c_glib/plasma-glib/client.cpp (hunk 1 above)
Comment (kou) [reply c_glib/plasma-glib/client.cpp]: No.
`shared_ptr` is just not needed here. We don't need to share a `plasma::PlasmaClient` created by `plasma_client_new()`.

File: c_glib/plasma-glib/client.cpp (hunk 2 above)
Comment (kou) [reply c_glib/plasma-glib/client.cpp]: Yes.
I forgot `return NULL` after `g_set_error()`.

File: c_glib/plasma-glib/object.cpp (hunk 3 above)
Comment (kou) [reply c_glib/plasma-glib/object.cpp]: Yes.
I've added it.

File: c_glib/plasma-glib/client.cpp (hunk 1 above)
Comment (shiro615) [reply c_glib/plasma-glib/client.cpp]: I see.

**Review by shiro615 — APPROVED** [review]
LGTM
I'll merge this.

**Pull request closed without merging**